.github/workflows/tests.yaml

```diff
@@ -353,6 +359,45 @@ jobs:
           path: "coverage.xml"
           retention-days: 1
 
+  tests-mlp2:
+    strategy:
+      fail-fast: true
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+          cache: pip
+          cache-dependency-path: tests/requirements**.txt
```

Comment:
you could consider using [`astral-sh/setup-uv`](https://github.com/astral-sh/setup-uv?tab=readme-ov-file#faq) in place of `pip install uv` below

Reply:
Ah, good to know! Thanks!

Reply:
> Using actions/setup-python can be faster, because GitHub caches the Python versions alongside the runner.

That part seems not ideal. Unclear to me what the difference is between that and `enable-cache: true` with `setup-uv` though.

Reply:
`enable-cache: true` refers to the package cache whereas that comment is talking about caching Python itself. you can use both actions together which is what I usually do. haven't tried `setup-uv` on its own though, could be that Python caching makes little difference